tla-types/src/main/scala/at/forsyte/apalache/tla/typecheck/etc/EtcTypeChecker.scala

```diff
@@ -170,7 +170,7 @@ class EtcTypeChecker(varPool: TypeVarPool, inferPolytypes: Boolean = false) exte
           onTypeFound(name.sourceRef, nameType)
           computeRec(ctx, solver, mkApp(ex.sourceRef, Seq(nameType), args: _*))
         } else {
-          onTypeError(ex.sourceRef, s"Undefined operator name $name. Introduce a type annotation.")
+          onTypeError(ex.sourceRef, s"It looks like the operator $name is used before it is defined. Is it true?")
```

Comment:
I think this is too vague and uncertain. Can we just say:

```suggestion
          onTypeError(ex.sourceRef, s"The operator $name is used before it is defined.")
```

Reply:
Well, I softened it a bit, because there was a bug where apalache barked on the operator order, while the bug was in apalache :D

Reply:
Errors happen. But I don't think it'll help the stability or usability of the tool in the long run if we phrase all the error messages tentatively in case we actually made a mistake in the implementation. It means we're optimizing for a pathological condition. IMO, if you really want to add a note communicating that we're not sure whether the implementation is correct in this error message, then you could just say that: "This detection may be in error. Please file an issue if you think it is." Or something more direct.